driver_test.go

```diff
@@ -2608,13 +2608,18 @@ func TestContextCancelBegin(t *testing.T) {
 	runTests(t, dsn, func(dbt *DBTest) {
 		dbt.mustExec("CREATE TABLE test (v INTEGER)")
 		ctx, cancel := context.WithCancel(context.Background())
-		tx, err := dbt.db.BeginTx(ctx, nil)
+		conn, err := dbt.db.Conn(ctx)
+		if err != nil {
+			dbt.Fatal(err)
+		}
+		defer conn.Close()
+		tx, err := conn.BeginTx(ctx, nil)
 		if err != nil {
 			dbt.Fatal(err)
 		}
 
 		// Delay execution for just a bit until db.ExecContext has begun.
-		defer time.AfterFunc(100*time.Millisecond, cancel).Stop()
+		go time.AfterFunc(100*time.Millisecond, cancel)
```

Comment:
Why this change?

Reply:
I actually didn't understand how this was working at all - what it _looked_ like it should be doing was canceling the context whilst the SQL was executing `SLEEP(1)`, so I turned it into this.

I just read the docs for `time.AfterFunc` more carefully and I think we can just invoke it on the main goroutine instead of calling `go`?

Reply:
Eurgh - ok the problem is I just don't know how defer works. This is going to cause `time.AfterFunc()` to run _now_, and then `Stop()` to run when the method returns. I'll change it back.